### PR TITLE
Fix module clashing

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Lots of minor changes at this point. See the Changelog.

At this point, there a few more changes are needed in the generateComponentConfig script, but I am thinking site is close to stable after these changes!

Future enhancements would be to enhance the site.config to allow for dynamic creation of the componentConfiguration such that there would not be a need to run the generateComponentConfig script before build.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,22 @@ Changelog
 
 Unreleased
 ----------
+### Added
+* Add support in generateComponentConfig for creating configuration from a terra repository that was installed as a package
+
+### Changed
+* Update webpack.config to only pass one
```
Comment (mjhenkes) [inline CHANGELOG.md]: only pass one what?

File: scripts/generate-component-config/build-component-config.js [hunk 2]
```diff
@@ -3,6 +3,20 @@ const fs = require('fs');
 const kebabCase = require('lodash.kebabcase');
 const startCase = require('lodash.startcase');
 
+const blackListedDirectories = [
```
Comment (mjhenkes) [inline scripts/generate-component-config/build-component-config.js]: blacklist and terraNameSpacedRepos seem like good candidates to pull out into a separate config. js, json, yml something like that.

File: scripts/generate-component-config/write-component-config.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ const writeComponentConfig = (packageConfigs, componentImports, outputPath, addP
     generatedConfig = generatedConfig.replace('<test_imports>', imports.tests);
     generatedConfig = generatedConfig.replace('<config>', `const componentConfig = ${configString};`);
 
-    fs.writeFileSync(path.join(path.resolve(process.cwd(), outputPath), 'generatedComponentConfig.js'), generatedConfig);
+    fs.writeFileSync(path.join(path.resolve(process.cwd(), outputPath), 'generatedComponentConfig.jsx'), generatedConfig);
```
Comment (mjhenkes) [inline scripts/generate-component-config/write-component-config.js]: Does the generated component config actually have jsx in it?

Comment (emilyrohrbough) [reply scripts/generate-component-config/write-component-config.js]: The generateComponentConfig output: https://github.com/cerner/terra-framework/blob/b0ad88a3f3d01025f2582c8dca748485d6db2526/generatedComponentConfig.jsx

File: CHANGELOG.md (hunk 1 above)
Comment (emilyrohrbough) [reply CHANGELOG.md]: Globally Defined Variable. ^ I'll update

File: scripts/generate-component-config/write-component-config.js (hunk 3 above)
Comment (bjankord) [reply scripts/generate-component-config/write-component-config.js]: Looks like it does not.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,22 @@ Changelog
 
 Unreleased
 ----------
+### Added
+* Add support in generateComponentConfig for creating configuration from a terra repository that was installed as a package
```
Comment (bjankord) [inline CHANGELOG.md]: 👍

File: scripts/generate-component-config/build-configs.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const terraNameSpacedRepos = [
+  'terra-core',
```
Comment (bjankord) [inline scripts/generate-component-config/build-configs.js]: Should terra-clinical be added to this list?

File: src/config/site.config.jsx [hunk 6]
```diff
@@ -21,9 +21,6 @@ module.exports = {
     /* The title the site header should display. */
     title: '', // maps to appTitle
 
-    /* The subtitle the site header should display. */
-    subtitle: '', // maps to appSubtitle
-
     /** The themes to supply the ThemeProvider which allows the site to switch
```
Comment (bjankord) [inline src/config/site.config.jsx]: This file doesn't look like it has any JSX in it, why the change from `.js` to `.jsx`?

File: CHANGELOG.md (hunk 4 above)
Comment (emilyrohrbough) [reply CHANGELOG.md]: @bjankord I should clarify, at this time, we still need to add custom search patterns to generate this config, however folders like src, tests, components, etc will automatically be filtered out

File: scripts/generate-component-config/build-configs.js (hunk 5 above)
Comment (emilyrohrbough) [reply scripts/generate-component-config/build-configs.js]: No it doesn't need to be. terra-core and terra-framework components are named 'terra-component-name' where as terra-clinical's are name-spaced 'terra-clinical-component-name'.

File: src/config/site.config.jsx (hunk 6 above)
Comment (emilyrohrbough) [reply src/config/site.config.jsx]: It allows us to import non-compiled files, so 

before I was dynamically importing the [`geneartedComponentConfig file`](https://github.com/cerner/terra-site/pull/15/files#diff-a7b1a86dc64b7d308e86bbc8f33e449aL26) in `Index.jsx`. This worked fine, however when I started working on terra-ui.com I realized I had removed the ability to add custom home, page component and menu components in the navigation.config. For example, see [here](https://github.com/cerner/terra-ui.com/compare/add-core-and-framework#diff-bc22fb7ebfd46781a950719ac185b30dR18).


With this chance we can directly pass the generatedComponentConfig and navigation config as objects and we reduce the need to [check for multiple file config and/or provide a defaults](https://github.com/cerner/terra-site/pull/15/files#diff-ed7a1de5bf46235874aea14f774f5b51L26) otherwise and it allows us to reduce the number of [globally defined variables from 2 to 1](https://github.com/cerner/terra-site/pull/15/files#diff-ed7a1de5bf46235874aea14f774f5b51L106).

File: scripts/generate-component-config/write-component-config.js (hunk 3 above)
Comment (mjhenkes) [reply scripts/generate-component-config/write-component-config.js]: Lets leave any file without jsx syntax in it marked as .js

File: src/config/site.config.jsx (hunk 6 above)
Comment (emilyrohrbough) [reply src/config/site.config.jsx]: 😑  well... after writing this, then seeing your comment above I went and changed some things in terra-ui. I am going to investigate a bit further because it seems this might not actually need to be changed.

**Comment by emilyrohrbough** [comment]
@bjankord @mjhenkes I changed generateComponentConfig, navigation.config, and site.config back to being .js files
